interfaces/builtin/socketcan.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const socketCanSummary = `allows use of SocketCAN network interfaces`
+
+const socketCanBaseDeclarationSlots = `
+  socketcan:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const socketCanConnectedPlugAppArmor = `
+# Description: Can access SocketCAN networks
+network can,
+`
+
+const socketCanConnectedPlugSecComp = `
+# Description: Can access SocketCAN networks.
+bind
+`
+
+func init() {
+	registerIface(&commonInterface{
+		name:                  "socketcan",
+		summary:               socketCanSummary,
+		implicitOnCore:        true,
+		implicitOnClassic:     true,
```

Comment:
We can't do it only on gadgets until we have dynamic slots (which are currently being worked on). Note that CONFIG_CAN is enabled in the Ubuntu 18.04 LTS kernel, so it is technically available on classic. I think it is fine to expose implicitly both places (particularly since it is manually connected).